### ISSUE TITLE
[question] Sign Change of Normalized Advantage Function in PPO2

### ISSUE THREAD

**Issue by yutingsz** [hill-a/stable-baselines#638]
The line 269 of the code "stable_baselines/ppo2/ppo2.py" normalizes the advantage as below
```advs = (advs - advs.mean()) / (advs.std() + 1e-8)```
My concern is this normalization method may change the sign of the original advantage value. Say the positive advantage value can become negative, if the value is less than the mean. Does this have an impact on the loss function? At this stage, I am not sure about the impact. Maybe this is fine.

**Comment by Miffyli** [comment]
This is the de-facto way of doing normalization. It is true half of the "advantage mass" will then have negative sign (and thus discourage taking those actions), but the overall effect is helpful for learning or sometimes outright necessary, especially if magnitudes of advantages are high (and thus the variance of losses is high, which quickly makes training unstable).

For a very-roughly-related topic on RL, check out "baselines" for policy gradients (e.g. this [Spinning Up section](https://spinningup.openai.com/en/latest/spinningup/rl_intro3.html#baselines-in-policy-gradients)). [GAE paper](https://arxiv.org/abs/1506.02438) also touches the topic tangentially by writing out different forms of advantages/returns you could use.
